server/utils/prisma/prisma-client.ts

```diff
@@ -1,7 +1,241 @@
-import { PrismaClient } from "@prisma/client";
+import { PrismaClient, Prisma } from "@prisma/client";
 
-const globalForPrisma = global as unknown as { prisma: PrismaClient };
+// Global client cache for initialized connections
+const clientCache = new Map<string, {
+  client: PrismaClient;
+  initialized: boolean; // Track if JWT claims are set
+}>();
 
-export const prisma = globalForPrisma.prisma || new PrismaClient();
+// Base Prisma client (singleton) for unauthenticated operations
+const globalForPrisma = global as unknown as { 
+  prisma: PrismaClient 
+};
+
+export const prisma = globalForPrisma.prisma || new PrismaClient({
+  log: ['query', 'error', 'warn'],
+  datasources: {
+    db: {
+      url: process.env.POSTGRES_URL_NON_POOLING,
+    },
+  },
+});
 
 if (process.env.NODE_ENV !== "production") globalForPrisma.prisma = prisma;
+
+// Create an authenticated client with JWT claims
+export function createAuthenticatedPrismaClient(userId: string) {
+  console.log(`Getting authenticated Prisma client for user ${userId}`);
+  
+  // Check if we have a cached client that's fully initialized
+  const cached = clientCache.get(userId);
+  if (cached?.initialized) {
+    console.log(`Using cached connection for user ${userId}`);
+    return applyExtensions(cached.client, userId);
+  }
+  
+  // Create a new PrismaClient with non-pooled connection
+  console.log(`Creating new connection for user ${userId}`);
+  const userClient = new PrismaClient({
+    log: ['query', 'error', 'warn'],
+    datasources: {
+      db: {
+        url: process.env.POSTGRES_PRISMA_URL,
+      },
+    },
+  });
+  
+  // Store in cache as uninitialized
+  clientCache.set(userId, {
+    client: userClient,
+    initialized: false
+  });
+  
+  // Create extended client with middleware that ensures JWT claims are set
+  const extendedClient = userClient.$extends({
+    name: 'auth-context',
+    query: {
+      $allModels: {
+        async $allOperations({ args, query, model, operation }) {
+          try {
+            // Ensure JWT claims are set before executing any query
+            await ensureJwtClaimsSet(userClient, userId);
+            
+            // Special handling for wallet searches
```

Comment:
Is this special handling here just for debugging? (as nothing is done with the result other than logging it):

Reply:
ah yeap.  Removed it.  Was only for debugging purposes.